src/Scheduler.ino

```diff
@@ -10,6 +10,7 @@
 #define PLUGIN_TASK_TIMER    2
 #define TASK_DEVICE_TIMER    3
 #define GPIO_TIMER           4
+#define PLUGIN_TIMER         5
```

Comment:
There is a PLUGIN_TASK_TIMER.
What does the PLUGIN_TIMER do?
The name is non descriptive and when is it called?

Its code looks like 99% copy of the PLUGIN_TASK_TIMER, so why not use that one with an extra parameter if needed?

It looks like it does act on plugin ID instead of taskIndex.
But we don't act on plugins, we act on tasks.

I will continue in the normal comments section.